contrib/checkbox-ce-oem/checkbox-ce-oem-snap/series_classic20/snap/snapcraft.yaml

```
@@ -74,6 +74,26 @@ parts:
       PART_SRC=${SNAPCRAFT_PROJECT_DIR}/parts/checkbox-provider-ce-oem/src
       gcc ${PART_SRC}/src/input-pcspkr.c -o ${PART_SRC}/src/beep
       cp -v ${PART_SRC}/src/beep ${SNAPCRAFT_PART_INSTALL}/usr/bin/
+  xtest:
+    plugin: dump
+    after: [checkbox-provider-ce-oem]
+    source: https://github.com/OP-TEE/optee_test.git
+    source-type: git
+    build-snaps:
+      - checkbox22
+    build-packages:
+      - jq
+    override-build: |
+      part_dest=${SNAPCRAFT_PART_INSTALL}/providers/checkbox-provider-ce-oem/data/
+      mkdir -p ${part_dest}
+      export PYTHONPATH=${SNAPCRAFT_PROJECT_DIR}/parts/checkbox-provider-ce-oem/src/bin/:/snap/checkbox22/current/lib/python3.10/site-packages/
+      source_file=${SNAPCRAFT_PROJECT_DIR}/parts/checkbox-provider-ce-oem/src/data/xtest_supported_version.json
+      script_file=${SNAPCRAFT_PROJECT_DIR}/parts/checkbox-provider-ce-oem/src/bin/optee_helper.py
```

Move this env variable to `build-environment` will be more readable
ref: https://snapcraft.io/docs/snapcraft-yaml-schema#p-21225-parts


```suggestion
    build-environment:
  	- part_dest: ${SNAPCRAFT_PART_INSTALL}/providers/checkbox-provider-ce-oem/data/
  	- PYTHONPATH: ${SNAPCRAFT_PROJECT_DIR}/parts/checkbox-provider-ce-oem/src/bin/:/snap/checkbox22/current/lib/python3.10/site-packages/
  	- source_file: ${SNAPCRAFT_PROJECT_DIR}/parts/checkbox-provider-ce-oem/src/data/xtest_supported_version.json
  	- script_file: ${SNAPCRAFT_PROJECT_DIR}/parts/checkbox-provider-ce-oem/src/bin/optee_helper.py
    override-build: |
```

---

thanks for the suggestions, I have updated my PR with the way you suggested.